sdk/core/core/test/cmocka/test_az_http.c

```diff
@@ -357,8 +358,160 @@ ENABLE_PRECONDITION_CHECK_TESTS()
+  }
+}
+
+static void test_http_request_header_validation_above_127(void** state)
+{
+  (void)state;
+  {
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_STR("some.url.com");
+    az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+    _az_http_request hrb;
+
+    TEST_EXPECT_SUCCESS(az_http_request_init(
+        &hrb,
+        &az_context_app,
+        az_http_method_get(),
+        url_span,
+        az_span_size(url_span),
+        header_span,
+        AZ_SPAN_FROM_STR("body")));
+
+    uint8_t c[1] = { 255 };
+    az_span header_name = AZ_SPAN_FROM_BUFFER(c);
+    ASSERT_PRECONDITION_CHECKED(
+        az_http_request_append_header(&hrb, header_name, hrb_header_content_type_token));
+
+    // make sure about header was not added
+    assert_int_equal(az_http_request_headers_count(&hrb), 0);
+    _az_http_request_headers headers = hrb._internal.headers;
+    size_t size = (size_t)az_span_size(headers);
+    assert_memory_equal(header_buf, headers._internal.ptr, size);
+  }
+}
+
 #endif // AZ_NO_PRECONDITION_CHECKING
 
+static void test_http_request_header_validation_range(void** state)
+{
+  (void)state;
+  {
+    // just make sure compiler will set anything greater than 127 to zero
+    for (int32_t value = 127; value < 256; value++)
+    {
+      assert_false(az_http_valid_token[(uint8_t)value]);
+    }
+  }
+}
+
+static void test_http_response_header_validation(void** state)
+{
+  (void)state;
+  {
+    az_http_response response = { 0 };
+    assert_return_code(
+        az_http_response_init(
+            &response,
+            AZ_SPAN_FROM_STR("HTTP/1.1 404 Not Found\r\n"
+                             "Header11: Value11\r\n"
+                             "Header22: NNNNOOOOPPPPQQQQRRRRSSSSTTTTUUUUVVVVWWWWXXXXYYYYZZZZ\r\n"
+                             "Header33:\r\n"
+                             "Header44: cba888888777777666666555555444444333333222222111111\r\n"
+                             "\r\n"
+                             "KKKKKJJJJJIIIIIHHHHHGGGGGFFFFFEEEEEDDDDDCCCCCBBBBBAAAAA")),
+        AZ_OK);
+
+    az_http_response_status_line status_line = { 0 };
+    assert_return_code(az_http_response_get_status_line(&response, &status_line), AZ_OK);
+    for (az_pair header;
+         az_http_response_get_next_header(&response, &header) != AZ_ERROR_ITEM_NOT_FOUND;)
+    {
+      // all valid headers
+      assert_true(az_span_ptr(header.key) != NULL);
+      assert_true(az_span_size(header.key) > 0);
+      assert_true(az_span_ptr(header.value) != NULL);
+    }
+  }
+}
+
+static void test_http_response_header_validation_fail(void** state)
+{
+  (void)state;
+  {
+    az_http_response response = { 0 };
+    assert_return_code(
+        az_http_response_init(
+            &response,
+            AZ_SPAN_FROM_STR("HTTP/1.1 404 Not Found\r\n"
+                             "(Header11): Value11\r\n"
```

Comment:
Add test where the header name contains a white space (for example `header name: value`). I am not certain, but I think we'd want to reject that. Please double check.